pkg/gather/collect/collect.go

```diff
@@ -446,6 +447,59 @@ func (c *Collector) collectNamespace(ctx context.Context, u *unstructured.Unstru
 	return nil
 }
 
+func (c *Collector) collectNamespaceForObject(ctx context.Context, u *unstructured.Unstructured, resourceInfo *ResourceInfo) error {
+	err := c.CollectResource(
+		ctx,
+		&ResourceInfo{
+			Scope:    meta.RESTScopeRoot,
+			Resource: corev1.SchemeGroupVersion.WithResource("namespaces"),
+		},
+		"",
+		u.GetNamespace(),
+	)
+	if err != nil {
+		return fmt.Errorf("can't collect namespace %q: %w", resourceInfo, err)
+	}
+
+	return nil
+}
+
+func (c *Collector) collectScyllaCluster(ctx context.Context, u *unstructured.Unstructured, resourceInfo *ResourceInfo) error {
+	err := c.writeResource(ctx, u, resourceInfo)
+	if err != nil {
+		return fmt.Errorf("can't write resource %q: %w", resourceInfo, err)
+	}
+
+	if !c.relatedResources {
+		return nil
+	}
+
+	err = c.collectNamespaceForObject(ctx, u, resourceInfo)
```

Comment:
given the func is called `collectScyllaCluster`, I'd move conditional collection of related resources to the switch case

Reply:
It's not a free function but member of a struct that has a field called `relatedResources`, so the member functions are supposed to account for them. Also in line with existing https://github.com/scylladb/scylla-operator/blob/5c354cea3a521661116f9d5f80a1bd62a5724cf4/pkg/gather/collect/collect.go#L403-L413

Reply:
ok